devicepluginmodbuscommander.cpp

```diff
@@ -0,0 +1,528 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ *                                                                         *
+ *  Copyright (C) 2019 Bernhard Trinnes <bernhard.trinnes@nymea.io>        *
+ *                                                                         *
+ *  This file is part of nymea.                                            *
+ *                                                                         *
+ *  This library is free software; you can redistribute it and/or          *
+ *  modify it under the terms of the GNU Lesser General Public             *
+ *  License as published by the Free Software Foundation; either           *
+ *  version 2.1 of the License, or (at your option) any later version.     *
+ *                                                                         *
+ *  This library is distributed in the hope that it will be useful,        *
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of         *
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU      *
+ *  Lesser General Public License for more details.                        *
+ *                                                                         *
+ *  You should have received a copy of the GNU Lesser General Public       *
+ *  License along with this library; If not, see                           *
+ *  <http://www.gnu.org/licenses/>.                                        *
+ *                                                                         *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+/*!
+    \page modbuscommander.html
+    \title Modbus Commander
+    \brief Plugin to write and read Modbus TCP coils and registers.
+
+    \ingroup plugins
+    \ingroup nymea-plugins
+
+
+    This plugin allows to write and read modbus registers and coils.
+
+    \underline{NOTE}: the library \c libmodbus has to be installed.
+
+    \chapter Plugin properties
+    Following JSON file contains the definition and the description of all available \l{DeviceClass}{DeviceClasses}
+    and \l{Vendor}{Vendors} of this \l{DevicePlugin}.
+
+    For more details how to read this JSON file please check out the documentation for \l{The plugin JSON File}.
+
+    \quotefile plugins/deviceplugins/modbuscommander/devicepluginmodbuscommander.json
+*/
+
+#include "devicepluginmodbuscommander.h"
+#include "plugininfo.h"
+
+#include <QSerialPort>
+
+DevicePluginModbusCommander::DevicePluginModbusCommander()
+{
+}
+
+void DevicePluginModbusCommander::init()
+{
+    connect(this, &DevicePluginModbusCommander::configValueChanged, this, &DevicePluginModbusCommander::onPluginConfigurationChanged);
```

Comment:
@mzanetti we might want to implement a method for this in the plugin api, if not already there, so the plugin can be reeimplemt it with override.

Reply:
hmm, yeah, maybe...